ui/src/pages/configurations/configuration/AdvancedConfigDialog.tsx

```diff
@@ -0,0 +1,182 @@
+import { UpdateStatus } from "../../../types/resources";
+import { isFunction } from "lodash";
+import { PERIODS } from "../../../components/MeasurementControlBar/MeasurementControlBar";
+
+interface Props extends DialogProps {
+  onSuccess: (interval: string) => void;
+  configName: string;
+}
+
+export const AdvancedConfigDialog: React.FC<Props> = ({
+  onSuccess,
+  configName,
+  ...dialogProps
+}) => {
+  const [measurementInterval, setMeasurementInterval] = useState("");
+  const [touched, setTouched] = useState(false);
+
+  const { enqueueSnackbar } = useSnackbar();
+
+  const { data, refetch } = useGetConfigurationQuery({
+    variables: {
+      name: configName,
+    },
+    onError(error) {
+      console.error(error);
+      enqueueSnackbar(`Failed to fetch configuration ${configName}.`, {
+        variant: "error",
+      });
+    },
+    onCompleted(data) {
+      if (data.configuration == null) {
+        enqueueSnackbar(`No configuration with name ${configName}.`, {
+          variant: "error",
+        });
+        return;
+      }
+    },
+  });
+
+  function clearState() {
+    setTouched(false);
+    setMeasurementInterval("");
+  }
+
+  async function updateMeasurementInterval() {
+    await refetch();
+    if (!data?.configuration) {
+      console.error("No configuration data to apply.");
+      return;
+    }
+
+    const updatedConfig = new BPConfiguration(data?.configuration);
+    updatedConfig.updateMeasurementInterval(measurementInterval + "s");
+    try {
+      const update = await updatedConfig.apply();
+      if (update.status === UpdateStatus.INVALID) {
+        console.error(update);
+        throw new Error("failed to add source to configuration.");
+      }
+
+      await refetch();
+      onSuccess(measurementInterval);
+      return;
+    } catch (err) {
+      console.error(err);
+      return;
+    }
+  }
+
+  async function handleSave(e: React.FormEvent<HTMLFormElement>) {
+    e.preventDefault();
+
+    try {
+      updateMeasurementInterval();
+
+      enqueueSnackbar("Measurements interval updated successfully", {
+        variant: "success",
+      });
+
+      // Call the provided onSuccess function
+      if (typeof onSuccess === "function") {
+        onSuccess(measurementInterval);
+      }
+    } catch (error) {
+      enqueueSnackbar("Failed to update Measurements interval", {
+        variant: "error",
+      });
+    }
+  }
+
+  function handleChange(
+    e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+  ) {
+    if (!touched) {
+      setTouched(true);
+    }
+    setMeasurementInterval(e.target.value);
+  }
+
+  return (
+    <Dialog
+      {...dialogProps}
+      TransitionProps={{
+        onExited: clearState,
+      }}
+    >
+      <DialogContent>
+        <Typography variant="h6" marginBottom={2}>
+          Advanced Configuration Options
+        </Typography>
+        <Typography>
+          Here, you can set the scrape interval for the agent's measurements.
```

Comment:
Comma might not be necessary, don't feel strongly about it.
```suggestion
          Here you can set the scrape interval for the agent's measurements.
```

Reply:
Updated